hermes-client/src/main/java/pl/allegro/tech/hermes/client/metrics/MicrometerMetricsProvider.java

```diff
@@ -17,30 +18,40 @@ public MicrometerMetricsProvider(MeterRegistry metrics) {
     }
 
     @Override
-    public void counterIncrement(String name) {
-        metrics.counter(name).increment();
+    public void counterIncrement(String topic, String key) {
+        counterIncrement(topic, key, new HashMap<>());
     }
 
     @Override
-    public void counterIncrement(String prefix, String name, Map<String, String> tags) {
-        metrics.counter(buildCounterName(prefix, name, tags), Tags.of(tags.entrySet().stream()
+    public void counterIncrement(String topic, String key, Map<String, String> tags) {
+        metrics.counter(buildCounterName(topic, key, tags), Tags.of(tags.entrySet().stream()
                .map(e -> Tag.of(e.getKey(), e.getValue()))
                .collect(Collectors.toSet())))
                .increment();
     }
 
     @Override
-    public void timerRecord(String name, long duration, TimeUnit unit) {
-        metrics.timer(name).record(duration, unit);
+    public void timerRecord(String topic, String key, long duration, TimeUnit unit) {
+        Map<String, String> tags = new HashMap<>();
+        metrics.timer(buildCounterName(topic, key, tags), Tags.of(tags.entrySet().stream()
+                .map(e -> Tag.of(e.getKey(), e.getValue()))
+                .collect(Collectors.toSet())))
+                .record(duration, unit);
     }
 
     @Override
-    public void histogramUpdate(String name, int value) {
-        metrics.summary(name).record(value);
+    public void histogramUpdate(String topic, String key, int value) {
+        Map<String, String> tags = new HashMap<>();
+        metrics.summary(buildCounterName(topic, key, tags), Tags.of(tags.entrySet().stream()
+                .map(e -> Tag.of(e.getKey(), e.getValue()))
+                .collect(Collectors.toSet())))
+                .record(value);
     }
 
-    private String buildCounterName(String prefix, String name, Map<String, String> tags) {
-        return prefix + "." + name + "."
-                + tags.keySet().stream().map(e -> "{" + e + "}").collect(Collectors.joining("."));
+    private String buildCounterName(String topic, String key, Map<String, String> tags) {
+        tags.put("topic", topic);
+        return prefix + "{topic}" + "." + key
```

Comment:
Have you tested it with graphite and without our internal SDK? As far as I know micrometer doesn't support this notation with placeholders. I see it was also used in this class before your change but I suspect it was a bug.

For more details, please take look at this issue: https://github.com/micrometer-metrics/micrometer/issues/620

Reply:
You are right, that wouldn't work without proper placeholders implementation in internal SDK. As You said, it was used here before for almost 2 years. There's also `DropwizardMetricProvider` implementation, which don't use this notation so it can be also used by clients, whose aren't familiar with placeholders.